include/pybind11/pybind11.h

```diff
@@ -827,18 +827,19 @@ class module : public object {
     PYBIND11_OBJECT_DEFAULT(module, object, PyModule_Check)
 
     /// Create a new top-level Python module with the given name and docstring
-    explicit module(const char *name, const char *doc = nullptr) {
-        if (!options::show_user_defined_docstrings()) doc = nullptr;
 #if PY_MAJOR_VERSION >= 3
-        PyModuleDef *def = new PyModuleDef();
+    explicit module(const char *name, const char *doc = nullptr, PyModuleDef *def = nullptr) {
+        if (!def) def = new PyModuleDef();
         std::memset(def, 0, sizeof(PyModuleDef));
         def->m_name = name;
```

Comment:
While we're at it, zeroing `def` and then not setting `def->base` is broken, according to [CPython docs](https://docs.python.org/3/c-api/module.html#c.PyModuleDef.m_base).

Reply:
I'm testing a new version, copying the commit message here for easy reference:

    Initializing PyModuleDef object with PyModuleDef_HEAD_INIT.
    
    Python 3.8 documentation: m_base - Always initialize this member to PyModuleDef_HEAD_INIT.
    
    Long-standing (since first github commit in 2015), inconsequential bug.
    
    Also removing inconsequential Py_INCREF(def): PyModule_Create() resets the reference count to 1.

I already tested locally with ASAN & MSAN. I'll report the result from the Google global tests when they are available.

Remark: I think the bug was inconsequential because the only non-zero member is the reference count, which is reset by `PyModule_Create()` anyway.